metatron/src/main/java/uk/gov/ida/eidas/metatron/domain/MetadataResolverService.java

```diff
@@ -1,39 +1,74 @@
+import uk.gov.ida.eidas.metatron.exceptions.MetatronServerException;
+import uk.gov.ida.notification.contracts.CountryMetadataResponse;
 import uk.gov.ida.saml.metadata.ExpiredCertificateMetadataFilter;
+import uk.gov.ida.saml.metadata.JerseyClientMetadataResolver;
 import uk.gov.ida.saml.metadata.PKIXSignatureValidationFilterProvider;
+import uk.gov.ida.saml.metadata.factories.CredentialResolverFactory;
 import uk.gov.ida.saml.metadata.factories.MetadataResolverFactory;
 
 import javax.ws.rs.client.Client;
+import javax.xml.namespace.QName;
+import java.net.URI;
+import java.security.cert.CertificateEncodingException;
 import java.util.ArrayList;
-import java.util.HashMap;
+import java.util.Base64;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 
 public class MetadataResolverService {
 
     protected static long MIN_REFRESH_DELAY_MS = 60_000;
     protected static long MAX_REFRESH_DELAY_MS = 600_000;
 
-    private Map<String, MetadataResolver> metadataResolverMap;
-    private ExpiredCertificateMetadataFilter expiredCertificateMetadataFilter;
-    private MetadataResolverFactory metadataResolverFactory;
+    private final Map<EidasCountryConfig, MetadataResolver> metadataResolverMap;
+    private final ExpiredCertificateMetadataFilter expiredCertificateMetadataFilter;
+    private final MetadataResolverFactory metadataResolverFactory;
+    private final CredentialResolverFactory credentialResolverFactory;
 
-    public MetadataResolverService(EidasConfig countriesConfig, MetadataResolverFactory metadataResolverFactory) {
+    public MetadataResolverService(
+            EidasConfig countriesConfig,
+            MetadataResolverFactory metadataResolverFactory,
+            CredentialResolverFactory credentialResolverFactory) {
         this.metadataResolverFactory = metadataResolverFactory;
-        this.metadataResolverMap = new HashMap<>();
+        this.credentialResolverFactory = credentialResolverFactory;
         this.expiredCertificateMetadataFilter = new ExpiredCertificateMetadataFilter();
-
-        metadataResolverMap = countriesConfig.getCountries().stream().collect(
-                Collectors.toMap(EidasCountryConfig::getEntityId, this::createMetadataResolver));
+        this.metadataResolverMap = countriesConfig.getCountries().stream()
+                .collect(Collectors.toMap(Function.identity(), this::createMetadataResolver));
     }
 
-    public MetadataResolver getMetadataResolver(String entityId) {
-        return metadataResolverMap.get(entityId);
+    public CountryMetadataResponse getCountryMetadataResponse(URI entityId) {
+        EidasCountryConfig countryConfig = getCountryConfig(entityId);
+        URI location = getAssertionConsumerServiceLocation(countryConfig);
+        String encryptionX509 = getCertificateAsX509(countryConfig, UsageType.ENCRYPTION);
+        String signingX509 = getCertificateAsX509(countryConfig, UsageType.SIGNING);
```

Comment:
These variable assignments could be dropped and the method calls inlined into the constructor.

Reply:
The entityId changes per request to this method, so they can't be inlined to constructor